support/sier-codec/src/builder.rs

```diff
@@ -0,0 +1,160 @@
+use crate::{FieldDef, Object, StructDef, Type, Value};
+
+use std::collections::HashMap;
+
+pub struct Builder<'s> {
+    struct_def: &'s StructDef,
+    field_values: HashMap<String, Value>,
+}
+
+impl<'s> Builder<'s> {
+    pub(crate) fn new(struct_def: &'s StructDef) -> Self {
+        Builder {
+            struct_def,
+            field_values: HashMap::new(),
+        }
+    }
+
+    pub fn set(mut self, field_name: &str, value: impl Into<Value>) -> Self {
+        self.field_values
+            .insert(field_name.to_string(), value.into());
+        self
+    }
+
+    pub fn try_build(mut self) -> Result<Object<'s>, BuildError> {
```

Comment:
Are we not adding the remaining two `build` methods?

Reply:
I don't want to do it right now. I just want to get to the point that you can start constructing and storing these. I don't think it will be a big effort to add them later.